### DISCUSSION TITLE
rev edge type confusion

### DISCUSSION THREAD

**Discussion by Klahadore** [pyg-team/pytorch_geometric#8712] [Q&A]
Hello
if I use ToUndirected(), will I need to manually add the edge attributes for the reverse edge, or will ToUndirected carry them over. 

Also for self attention, if is do something like Linear(torch.cat(x_i, x_i)), or do i define add self loops to the dataset, meaning i will also have to add edge attributes for self attention?
Thanks in advance

**Comment by rusty1s** [comment]
> if I use ToUndirected(), will I need to manually add the edge attributes for the reverse edge, or will ToUndirected carry them over.

They will be carried over.

> Also for self attention, if is do something like Linear(torch.cat(x_i, x_i)), or do i define add self loops to the dataset, meaning i will also have to add edge attributes for self attention?

Not totally sure I understand. If you define self-loops in your graph and your graph has edge features, then the self-loops should also define edge features.
